### DISCUSSION TITLE
zopen-config challenge

### DISCUSSION THREAD

**Discussion by lbdyck** [zopencommunity/meta#756] [General]
For `zigi` and other tools to get a list of the active environment variables using `bpxwunix` from TSO the need to run the `zopen-config` out of the `/etc/profile` is a problem as it does not appear to run, or if it does the environment variables it sets are not retained long enough for the `env` command to the standard output stem.

ex:  `export HOME=/u/lbdyck && . /etc/profile > /dev/null 2>&1 &&  . /u/lbdyck/.profile > /dev/null 2>&1 && env `

and the call:  ` rc = bpxwunix(cmd,,env.,se.,omvsenv.,1) `

where `omvsenv.` is set thus:

```
omvsenv.1 = '_BPX_SHAREAS=MUST'   
omvsenv.0 = 1                     
```

It would be ***very*** helpful if the `zopen` install/upgrade/remove/? could generate a file that can be appended into `/etc/profile` with the appropriate environment variable updates.

**Comment by lbdyck** [comment]
I have written a tool but it has it's issues and it would be better to be intergated.

**Comment by DevonianTeuchter** [comment]
Not sure I quite follow... Zopen-config doesn't have to to run from .profile, it can be run "standalone" so for instance:
export HOME=/u/lbdyck && . /etc/profile > /dev/null 2>&1 &&  . /u/lbdyck/.profile > /dev/null 2>&1 && . <Zopen-root>/etc/zopen-config && env
Does that work??

Not at my pc/terminal for a few days so can't test but it'd be interesting to find out if (and why iff) the nested "source"/. didn't work.
Similarly, copy'n'pasting the contents of zopen-config into .profile directly could also work - that just bypasses the sourcing step so nested sourcing does not work, the code running directly might. 
There's a new script "zopen-config" in the works that could have something like an export or env param to generate what you're after

**Comment by lbdyck** [comment]
Interesting idea of putting the `zopen-config` directly in the `/etc/profile` - as for trying to add `. /etc/zopen-config` to the command would probably fail as `zopen-config` may not be in `/etc`.  

An `export` or `env` parm sounds interesting.

Enjoy your time away.

**Comment by lbdyck** [comment]
After more experimentation I've learned

1. putting `zopen-config` into the `/etc/profile` works for shell and OMVS access but does **not** work when using the `bpxwunix` with the `login` option.
2. using the users `.profile` is the same as (1) and found the `.profile` is not used if the users shell is other than `sh` (i.e. `bash` uses `.bashrc`  and `zsh` uses `.zshrc`
3. found my work system uses `/global/etc/profile` which is different from `/etc/profile` and causes the `/etc/profile` to be ignored/bypassed

hope this helps a bit

I have updated `zigi` so that the environment file that it uses can be the `zopen-config` and that works - the ***BIG*** issue with that is that every `zigi` user has to specify it in their `zigi` config.

**Comment by lbdyck** [comment]
My `zotenv` tool works well to generate a set of `export` statements for use in `/etc/profile`, or ..., but there are some issues - six specifically with the limited set of tools installed via `zopen`.  The release specific references means that with each update the `/etc/profile`, or ..., needs to be updated after running the tool. 

Any chance of using clean aliases?

![image](https://github.com/ZOSOpenTools/meta/assets/42328411/e3e01ff6-89e1-4ae7-b667-0a40d366e163)

**Comment by DevonianTeuchter** [comment]
I could see a way to possibly circumvent items 1,2,3,5 &6 but 4 would be tricky (given that's baked into the pax itself) - the current version of a package has a symlink to it so potentially could use that instead ( eg. usr/local/zopen/groff/groff/ for example). Would need to see where those envvars get their setting from as I suspect it's in the setup scripts shipped in the pax and might need a bit of manipulation

**Reply by lbdyck** [reply]
any thoughts?
